pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -617,6 +624,40 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
     return prematureExit;
   }
 
+  private void updateConsumerLagMetrics() {
+    boolean emitRecordsLag = true;
+    boolean emitAvailabilityLag = true;
+    if (_caughtUpWithUpstream) {
```

Comment:
why has this condition? we should always publish the metric right?

Reply:
This is to handle the catch up scenario. If the partition is still "catching up" to the upstream, this metric will spike, even though it is not indicative of any issue with consumption. Thus, we wanted to avoid the potential for false alerts with this metric.

Reply:
It is ok for the metric to spike in the short term. But we need to know how far behind we are, and how fast we are catching up.